### ISSUE TITLE
Test suite should ignore build-id differences in binaries

### ISSUE THREAD

**Issue by rwmjones** [ocaml/ocaml#9826]
We have lots of failures in the test suite.  Check for "failed" in this log: https://kojipkgs.fedoraproject.org//packages/ocaml/4.11.0/0.8.dev2.fc33/data/logs/x86_64/build.log

Most of these are distractions.  They are caused because the binary produced by ocamlopt.byte doesn't exactly match the binary produced by ocamlopt.opt, and the cause of that is (I believe only) the build ID.  Build IDs are added downstream by Fedora and other distros to associate debug info with a particular binary.

For example here is an error about file differences:

 ... testing 'is_static.ml' with 1 (native) => failed (Files /builddir/build/BUILD/ocaml-4.11.0/testsuite/_ocamltest/tests/asmcomp/is_static/ocamlopt.byte/is_static.opt and /builddir/build/BUILD/ocaml-4.11.0/testsuite/_ocamltest/tests/asmcomp/is_static/ocamlopt.opt/is_static.opt are different)

But after stripping the binaries and removing the build ID note they are actually identical:

```
$ cp /var/tmp/oo/master/ocaml-4.11.0/testsuite/_ocamltest/tests/asmcomp/is_static/ocamlopt.byte/is_static.opt is_static.opt.byte
$ cp /var/tmp/oo/master/ocaml-4.11.0/testsuite/_ocamltest/tests/asmcomp/is_static/ocamlopt.opt/is_static.opt is_static.opt.opt
$ ll is_static.opt.*
-rwxr-xr-x. 1 rjones rjones 2113792 Aug  4 14:19 is_static.opt.byte
-rwxr-xr-x. 1 rjones rjones 2113776 Aug  4 14:19 is_static.opt.opt
$ strip --remove-section=.note.gnu.build-id is_static.opt.*
$ ll is_static.opt.*
-rwxr-xr-x. 1 rjones rjones 311240 Aug  4 14:19 is_static.opt.byte
-rwxr-xr-x. 1 rjones rjones 311240 Aug  4 14:19 is_static.opt.opt
$ cmp is_static.opt.*
$ echo $?
0
```

Looking in `ocamltest/` it seems as if the comparison program might be configurable in some way but I have no idea how to do it.

An alternative might be that when we build programs in the test suite we use `-Wl,--build-id=none` (which should be safe on all Linux distros, but assumes you're using GNU ld).

**Comment by nojb** [comment]
The testsuite was modified to stop comparing native-code binaries #9677, which will be included in 4.12.

**Comment by shindere** [comment]
I have no strong opinion about how this should best be delt with.

If the other developers feel that this comparison is a too strict test
it could also be disabled completely, I don't know.

**Comment by shindere** [comment]
Nicolás Ojeda Bär (2020/08/04 06:45 -0700):
> The testsuite was modified to stop comparing native-code binaries
> #9677, which will be included in 4.12.

Perhaps that should be backported to 4.11? @Octachron?

**Comment by rwmjones** [comment]
If we're no longer testing this upstream then feel free to close this bug.
